examples/parallel_memmap.py

```diff
@@ -38,44 +22,44 @@ def sum_row(input, output, i):
-        samples = rng.normal(size=(10, int(1e6)))
 
-        # Pre-allocate a writeable shared memory map as a container for the
-        # results of the parallel computation
-        sums = np.memmap(sums_name, dtype=samples.dtype,
-                         shape=samples.shape[0], mode='w+')
+rng = np.random.RandomState(42)
+folder = tempfile.mkdtemp()
+samples_name = os.path.join(folder, 'samples')
+sums_name = os.path.join(folder, 'sums')
+try:
+    # Generate some data and an allocate an output buffer
+    samples = rng.normal(size=(10, int(1e6)))
+
+    # Pre-allocate a writeable shared memory map as a container for the
+    # results of the parallel computation
+    sums = np.memmap(sums_name, dtype=samples.dtype,
+                     shape=samples.shape[0], mode='w+')
 
-        # Dump the input data to disk to free the memory
-        dump(samples, samples_name)
+    # Dump the input data to disk to free the memory
+    dump(samples, samples_name)
 
-        # Release the reference on the original in memory array and replace it
-        # by a reference to the memmap array so that the garbage collector can
-        # release the memory before forking. gc.collect() is internally called
-        # in Parallel just before forking.
-        samples = load(samples_name, mmap_mode='r')
+    # Release the reference on the original in memory array and replace it
+    # by a reference to the memmap array so that the garbage collector can
+    # release the memory before forking. gc.collect() is internally called
+    # in Parallel just before forking.
+    samples = load(samples_name, mmap_mode='r')
 
-        # Fork the worker processes to perform computation concurrently
-        Parallel(n_jobs=4)(delayed(sum_row)(samples, sums, i)
-                           for i in range(samples.shape[0]))
+    # Fork the worker processes to perform computation concurrently
+    Parallel(n_jobs=4)(delayed(sum_row)(samples, sums, i)
+                       for i in range(samples.shape[0]))
 
-        # Compare the results from the output buffer with the ground truth
-        print("Expected sums computed in the parent process:")
-        expected_result = samples.sum(axis=1)
-        print(expected_result)
+    # Compare the results from the output buffer with the ground truth
+    print("Expected sums computed in the parent process:")
+    expected_result = samples.sum(axis=1)
+    print(expected_result)
 
-        print("Actual sums computed by the worker processes:")
-        print(sums)
+    print("Actual sums computed by the worker processes:")
+    print(sums)
 
-        assert np.allclose(expected_result, sums)
-    finally:
-        try:
-            shutil.rmtree(folder)
-        except:
-            print("Failed to delete: " + folder)
+    assert np.allclose(expected_result, sums)
+finally:
+    try:
+        shutil.rmtree(folder)
+    except:  # noqa
+        print("Failed to delete: " + folder)
```

Comment:
I think there is no point in hiding the exception if `shutil.rmtree(folder)` fails. Let's remove the try / except around it.